pycparser/c_generator.py

```diff
@@ -72,11 +73,54 @@ def visit_UnaryOp(self, n):
             else:
                 return '%s%s' % (n.op, operand)
 
+    op_token_map = {
+        'PLUS': '+',
+        'MINUS': '-',
+        'TIMES': '*',
+        'DIVIDE': '/',
+        'MOD': '%',
+        'OR': '|',
+        'AND': '&',
+        'NOT': '~',
+        'XOR': '^',
+        'LSHIFT': '<<',
+        'RSHIFT': '>>',
+        'LOR': '||',
+        'LAND': '&&',
+        'LNOT': '!',
+        'LT': '<',
+        'GT': '>',
+        'LE': '<=',
+        'GE': '>=',
+        'EQ': '==',
+        'NE': '!='
+    }
+
+    @property
+    def _prededence_map(self):
+        """
+        Map of operands (e.g., '%') to precedence (higher number -> stronger precedence).
+        """
+        if not hasattr(self, '_prededence_map_cache'):
+            self._prededence_map_cache = {}
```

Comment:
I may have misled you here earlier, sorry. I'd actually prefer this not to depend on `c_parser` -- this seems like a needless dependency that can complicate things.

I see two options: either we leave this just duplicated in two places - it's not like the C operator precedence is going to change! Or we refactor it to a separate file which both `c_parser` and `c_generator` import. WDYT?

Reply:
I duplicated it and added a comment do `c_parser.py`